### PR TITLE
GH-35528:[Java] Fix RangeEqualsVisitor comparing BitVector with different begin index

### LINKED ISSUE #35528: [Java]bugfix: RangeEqualsVisitor compare BitVector false when compared vectors have different begin index.
### Describe the bug, including details regarding any error messages, version, and platform.

In origin code, when compared vectors have different begin index, it will compare vectors value with same index, which is error logic. this pr is to fix it.

### Component(s)

Java

### PR DESCRIPTION
**Pull request by wenxlan**
### Rationale for this change

bugfix: RangeEqualsVisitor compare BitVector false when compared vectors have different begin index. 

In origin code, when compared vectors have different begin index, it will compare vectors value with same index, which is error logic. this pr is to fix it.

### What changes are included in this PR?

only changes RangeEqualsVisitor.compareBaseFixedWidthVectors method, and add test

### Are these changes tested?
yes.

### Are there any user-facing changes?
no.

* Closes: #35528

### CODE REVIEW

File: java/vector/src/test/java/org/apache/arrow/vector/compare/TestRangeEqualsVisitor.java [hunk 1]
```diff
@@ -26,13 +26,7 @@
...
 import org.apache.arrow.memory.BufferAllocator;
 import org.apache.arrow.memory.RootAllocator;
-import org.apache.arrow.vector.BigIntVector;
-import org.apache.arrow.vector.Float4Vector;
-import org.apache.arrow.vector.Float8Vector;
-import org.apache.arrow.vector.IntVector;
-import org.apache.arrow.vector.LargeVarCharVector;
-import org.apache.arrow.vector.VarCharVector;
-import org.apache.arrow.vector.ZeroVector;
+import org.apache.arrow.vector.*;
```
Comment (lidavidm) [inline java/vector/src/test/java/org/apache/arrow/vector/compare/TestRangeEqualsVisitor.java:29]: nit: I realize this was probably the IDE but can this be reverted? For review it's easier to list out imports

**Review by lidavidm — COMMENTED** [review]
LGTM overall. Thanks! 

(Just one comment.)

File: java/vector/src/test/java/org/apache/arrow/vector/compare/TestRangeEqualsVisitor.java (hunk 1 above)
Comment (wenxlan) [reply java/vector/src/test/java/org/apache/arrow/vector/compare/TestRangeEqualsVisitor.java]: it is reverted now～

**Review by lidavidm — APPROVED** [review]
LGTM, will merge after CI

**Review by lidavidm — APPROVED** [review]
Thanks!

**Pull request merged**